segmentation_models_pytorch/encoders/timm_gernet.py

```diff
@@ -0,0 +1,120 @@
+from timm.models import ByobCfg, BlocksCfg, ByobNet
+
+from ._base import EncoderMixin
+import torch.nn as nn
+
+
+class GERNetEncoder(ByobNet, EncoderMixin):
+    def __init__(self, out_channels, depth=6, **kwargs):
+        super().__init__(**kwargs)
+        self._depth = depth
+        self._out_channels = out_channels
+        self._in_channels = 3
+
+        del self.head
+
+    def get_stages(self):
+        return [
+            nn.Identity(),
+            self.stem,
+            *self.stages[:-1],
+            nn.Sequential(self.stages[-1], self.final_conv)
+        ]
+
+    def forward(self, x):
+        stages = self.get_stages()
+
+        features = []
+        for i in range(self._depth + 1):
+            x = stages[i](x)
+            features.append(x)
+
+        return features
+
+    def load_state_dict(self, state_dict, **kwargs):
+        state_dict.pop("head.fc.weight")
+        state_dict.pop("head.fc.bias")
+        super().load_state_dict(state_dict, **kwargs)
+
+
+regnet_weights = {
+    'timm-gernet_s': {
+        'imagenet': 'https://github.com/rwightman/pytorch-image-models/releases/download/v0.1-ger-weights/gernet_s-756b4751.pth',
+    },
+    'timm-gernet_m': {
+        'imagenet': 'https://github.com/rwightman/pytorch-image-models/releases/download/v0.1-ger-weights/gernet_m-0873c53a.pth',
+    },
+    'timm-gernet_l': {
+        'imagenet': 'https://github.com/rwightman/pytorch-image-models/releases/download/v0.1-ger-weights/gernet_l-f31e2e8d.pth',
+    },
+}
+
+pretrained_settings = {}
+for model_name, sources in regnet_weights.items():
+    pretrained_settings[model_name] = {}
+    for source_name, source_url in sources.items():
+        pretrained_settings[model_name][source_name] = {
+            "url": source_url,
+            'input_size': [3, 224, 224] if not model_name == 'timm-gernet_l' else [3, 256, 256],
```

Comment:
Not sure that such if else inside builder is appropriate, but it seems better than 3 separate large configs

Reply:
`input_size` is not necessery kwarg, you may remove it